packages/floating-label/index.tsx

```diff
@@ -19,28 +19,41 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
+import * as React from 'react';
+import * as classnames from 'classnames';
+// no mdc .d.ts
+// @ts-ignore
+import {MDCFloatingLabelFoundation} from '@material/floating-label';
+
+export interface FloatingLabelProps extends React.LabelHTMLAttributes<HTMLLabelElement> {
+  className?: string;
+  handleWidthChange?: (width: number) => void;
+  float?: boolean;
+};
 
-import React from 'react';
-import PropTypes from 'prop-types';
-import classnames from 'classnames';
-import {MDCFloatingLabelFoundation} from '@material/floating-label/dist/mdc.floatingLabel';
+interface FloatingLabelState {
+  classList: Set<string>;
+};
 
-export default class FloatingLabel extends React.Component {
-  foundation_ = null;
+export default class FloatingLabel extends React.Component<
+  FloatingLabelProps,
+  FloatingLabelState
+  > {
+  foundation_?: MDCFloatingLabelFoundation;
+  labelElement: React.RefObject<HTMLLabelElement> = React.createRef();
```

Comment:
`labelElement_`? I don't know if we've been consistent with our use of underscores for refs

Reply:
ya we haven't, but since everything is already I will change this one. I think we should remove all of them eventually and use the `private` keyword if possible